ladybug_geometry/polyskel.py

```diff
@@ -0,0 +1,558 @@
+		_debug.line((self.bisector.p.x, self.bisector.p.y, self.bisector.p.x+self.bisector.v.x*100, self.bisector.p.y+self.bisector.v.y*100), fill="blue")
+
+	@property
+	def bisector(self):
+		return self._bisector
+
+	@property
+	def is_reflex(self):
+		return self._is_reflex
+
+	@property
+	def original_edges(self):
+		return self.lav._slav._original_edges
+
+	def next_event(self):
+		events = []
+		if self.is_reflex:
+			# a reflex vertex may generate a split event
+			# split events happen when a vertex hits an opposite edge, splitting the polygon in two.
+			log.debug("looking for split candidates for vertex %s", self)
+			for edge in self.original_edges:
+				if edge.edge == self.edge_left or edge.edge == self.edge_right:
+					continue
+
+				log.debug("\tconsidering EDGE %s", edge)
+
+				# A potential b is at the intersection of between our own bisector and
+				# the bisector of the angle between the tested edge and any one of our
+				# own edges.
+
+				# We choose the "less parallel" edge (in order to exclude a
+				# potentially parallel edge)
+
+				# Make normalized copies of vectors
+				norm_edge_left_v  = Vector2D.normalize(self.edge_left.v)
+				norm_edge_right_v = Vector2D.normalize(self.edge_right.v)
+				norm_edge_v = Vector2D.normalize(edge.edge.v)
+
+				# Compute dot
+				leftdot  = abs(norm_edge_left_v.dot(norm_edge_v))
+				rightdot = abs(norm_edge_right_v.dot(norm_edge_v))
+				selfedge = self.edge_left if leftdot < rightdot else self.edge_right
+				otheredge = self.edge_left if leftdot > rightdot else self.edge_right
+
+				# Make copies of edges and compute intersection
+				self_edge_copy = LineSegment2D(selfedge.p, selfedge.v)
+				edge_edge_copy = LineSegment2D(edge.edge.p, edge.edge.v)
+				i = intersection2d.intersect_line2d(self_edge_copy, edge_edge_copy)
+				if i is not None and not _approximately_equals(i, self.point):
+					#locate candidate b
+					linvec = Point2D.normalize(self.point - i)
+					edvec  = Point2D.normalize(edge.edge.v)
+					if linvec.dot(edvec)<0:
+						edvec = -edvec
+
+					bisecvec = edvec + linvec
+					if abs(bisecvec) == 0:
+						continue
+					bisector = LineSegment2D(i, bisecvec)
+					b = bisector.intersect(self.bisector)
+
+					if b is None:
+						continue
+
+					#check eligibility of b
+					# a valid b should lie within the area limited by the edge and the bisectors of its two vertices:
+					xleft = _determinant(edge.bisector_left.v.normalized(), (b - edge.bisector_left.p).normalized()) > 0
+					xright = _determinant(edge.bisector_right.v.normalized(), (b - edge.bisector_right.p).normalized()) < 0
+					xedge = _determinant(edge.edge.v.normalized(), (b - edge.edge.p).normalized()) < 0
+
+					if not (xleft and xright and xedge):
+						log.debug("\t\tDiscarded candidate %s (%s-%s-%s)", b, xleft, xright, xedge)
+						continue
+
+					log.debug("\t\tFound valid candidate %s", b)
+					events.append( _SplitEvent(Line2(edge.edge).distance(b), b, self, edge.edge) )
```

Comment:
Is this part uses 4 spaces? Looks like 8 to me.

Reply:
Yeap, I set this manually and Pycharm and just counted the space in this section manually it's 4. I think it just looks long because the first line is a condition.